content/en/docs/concepts/architecture/garbage-collection.md

```diff
@@ -95,8 +95,10 @@ to learn more.
 ### Background cascading deletion {#background-deletion}
 
 In background cascading deletion, the Kubernetes API server deletes the owner
-object immediately and the controller cleans up the dependent objects in
-the background. By default, Kubernetes uses background cascading deletion unless
+object immediately and the controller (like the custom or default controller)
+cleans up the dependent objects in the background.
+If a finalizer exists, it ensures that objects are not deleted until all necessary clean-up tasks are completed.
```

Comment:
This may be better

```suggestion
If a finalizer exists, it ensures the resources the deleted object owned are deleted before the object itself.
```

Reply:
Thanks for the suggestion, @reylejano. The suggestion confused me at first due to the duplicated “the” used at proximity. I’ll paraphrase it to “`If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.`”

Reply:
```suggestion
If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.
```

Reply:
A finalizer is a blocker and the list of finalizers is mutable by other objects which can be controller or something else. The finalizer is not necessarily linked to an owner.

Reply:
if a finalizer doesn't have to imply an ownership relation, isn't the revised description below problematic?

> If a finalizer exists, it ensures that resources owned by the deleted object are deleted before the object itself.

Reply:
It is. Maybe we could use the initial sentence I suggested to avoid confusion, as it's more general.

> If a finalizer exists, it ensures that objects are not deleted until all necessary clean-up tasks are completed.